src/indexApi.ts

```diff
@@ -1,18 +1,125 @@
 // BASE SETUP
 // =============================================================================
 
+import { asObject, asOptional, asString } from 'cleaners'
 import cluster from 'cluster'
+import { formatISO } from 'date-fns'
 import { forkChildren, rebuildCouch } from 'edge-server-tools'
 
 import { config } from './utils/config'
-import { couchSchema } from './utils/couchSchema'
+import { couchSchema, minAmtDbPrefix } from './utils/couchSchema'
+import { ErrorResponse, makeErrorResponse } from './utils/errorResponse'
+import { mockPlugins } from './utils/mockPlugins'
 
 const express = require('express')
 const http = require('http')
+const nano = require('nano')
+const promisify = require('promisify-node')
+
+const asFulfilledPromise = asObject({
+  status: asString,
+  value: asObject({
+    _id: asString,
+    _rev: asOptional(asString),
+    data: asObject({ minAmount: asString })
+  })
+})
+
+const GetMinimumParamError: ErrorResponse = makeErrorResponse(
+  'bad_request',
+  400,
+  'Missing query params'
+)
+
+const CurrencyPairDataError: ErrorResponse = makeErrorResponse(
+  'not_found',
+  404,
+  'Data for currencyPair not found'
+)
+
+const RouteError: ErrorResponse = makeErrorResponse(
+  'bad_query',
+  404,
+  'Endpoint not found'
+)
 
 // call the packages we need
 const app = express()
 
+// Nano for CouchDB
+// =============================================================================
+const nanoDb = nano(config.dbFullpath)
+const swapMinAmts = mockPlugins.map(plugin => {
+  const pluginMinAmt = nanoDb.db.use(minAmtDbPrefix + plugin.pluginName)
+  return promisify(pluginMinAmt)
+})
+
+// ROUTES FOR OUR API
+// =============================================================================
+const router = express.Router()
+
+router.use(function (req, res, next) {
+  console.log('Something is happening.')
+  next() // make sure we go to the next routes and don't stop here
+})
+
+router.get('/getMinimum/', async function (req, res, next) {
+  const currencyPair: string = req.query.currencyPair // Param of 'currencyPair' in URI
+  if (currencyPair === undefined) {
+    return next(GetMinimumParamError)
+  }
+  const currentUTCDate: string = formatISO(new Date(), {
+    representation: 'date'
+  }) // Variable for current UTC date using date-fns
+  try {
```

Comment:
Break this try..catch into testable utility functions. Middleware should have as little functions as possible.